### PR TITLE
Fix to increment/decrement gas-limit in block production 

### PR DESCRIPTION
**Pull request by non-fungible-nelson**
## PR description
Fixes discord bug found by protoplanetary

### CODE REVIEW

**Comment by siladu** [comment]
Think this is the geth code and it doesn't include a constant 1024 increment so besu does appear to be wrong according to both the spec and geth...https://github.com/ethereum/go-ethereum/blob/e5d5e09faae48dac3723634e2b1813e4f2e89535/core/block_validator.go#L152

**Review by siladu — APPROVED** [review]
LGTM

**Review by jflo — APPROVED** [review]
wow. just wow.

**Review by garyschulte — APPROVED** [review]
As long as we verified the arithmetic and the resulting [diff](https://github.com/hyperledger/besu/pull/6425/files#diff-082fa44dde5db1486ea62bda464aaba35e229a05a2b416f114157b09f7e9691dL13) in the test artifact, 🚢

**Comment by siladu** [comment]
@garyschulte 

> As long as we verified the arithmetic and the resulting [diff](https://github.com/hyperledger/besu/pull/6425/files#diff-082fa44dde5db1486ea62bda464aaba35e229a05a2b416f114157b09f7e9691dL13) in the test artifact, 🚢

currentGasLimit = 3141592 ([0x2fefd8](https://github.com/hyperledger/besu/blob/2d7941f8b84eb311a42b52270daf32e0465429ca/acceptance-tests/tests/src/test/resources/jsonrpc/engine/cancun/test-cases/block-production/09_cancun_get_chainhead.json#L31))
```
currentGasLimit = 3141592
targetGasLimit = 30000000 (not set so uses default)
3141592 // 1024 = 3067
3067 - 1 = 3066 (`maxProportionalAdjustmentLimit`)
3141592 + 3066 = 3144658

3144658 = 0x2FFBD2

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,7 @@
 - Fix Besu Docker images with `openjdk-latest` tags since 23.10.3 using UID 1001 instead of 1000 for the `besu` user [#6360](https://github.com/hyperledger/besu/pull/6360)
 - Fluent EVM API definition for Tangerine Whistle had incorrect code size validation configured [#6382](https://github.com/hyperledger/besu/pull/6382)
 - Correct mining beneficiary for Clique networks in TraceServiceImpl [#6390](https://github.com/hyperledger/besu/pull/6390)
+- Fix to gas limit delta calculations used in block production. Besu should now incrment or decerement the block gas limit towards its target correctly (thanks @arbora) #6425
```
Comment (macfarla) [inline CHANGELOG.md:29]:
```suggestion
- Fix to gas limit delta calculations used in block production. Besu should now increment or decrement the block gas limit towards its target correctly (thanks @arbora) #6425
```

**Comment by siladu** [comment]
Added same unit tests as geth for extra robustness https://github.com/ethereum/go-ethereum/blob/830f3c764c21f0d314ae0f7e60d6dd581dc540ce/core/block_validator_test.go#L242

**Pull request merged**